tests/cache/tests.py

```diff
@@ -1040,6 +1040,16 @@ def test_delete_many_num_queries(self):
     def test_zero_cull(self):
         self._perform_cull_test('zero_cull', 50, 18)
 
+    def test_cull_delete_when_table_empty(self):
+        try:
+            cull_cache = caches['cull']
+        except InvalidCacheBackendError:
+            self.skipTest("Culling isn't implemented.")
+
+        cull_cache._max_entries = -1  # Force _cull to delete on first cached record
+        cull_cache.set('force_cull_delete', 'value', 1000)
+        self.assertTrue(cull_cache.has_key('force_cull_delete'))
+
```

Comment:
Could this test be moved to `BaseCacheTests` under `test_empty_cull` to make sure the implementation works on all backends instead of only the database one?

https://github.com/django/django/blob/0bebe5266f2e52a76fcf6d23b76942399d087bf2/tests/cache/tests.py#L601-L622

Reply:
Sure I'll do it tonight and update the PR, thanks for the review

Reply:
Thanks @charettes for the review. Code moved to BaseCacheTests